assets/js/googlesitekit/widgets/util/get-widget-layout.js

```diff
@@ -19,49 +19,29 @@
 /**
  * Internal dependencies
  */
-import { WIDGET_WIDTHS } from '../datastore/constants';
-
-const WIDTH_GRID_COUNTER_MAP = {
-	[ WIDGET_WIDTHS.QUARTER ]: 3,
-	[ WIDGET_WIDTHS.HALF ]: 6,
-	[ WIDGET_WIDTHS.FULL ]: 12,
-};
-
-const WIDTH_GRID_CLASS_MAP = {
-	[ WIDGET_WIDTHS.QUARTER ]: [
-		'mdc-layout-grid__cell',
-		'mdc-layout-grid__cell--span-2-phone',
-		'mdc-layout-grid__cell--span-3-desktop',
-		'mdc-layout-grid__cell--span-4-tablet',
-	],
-	[ WIDGET_WIDTHS.HALF ]: [
-		'mdc-layout-grid__cell',
-		'mdc-layout-grid__cell--span-6-desktop',
-		'mdc-layout-grid__cell--span-8-tablet',
-	],
-	[ WIDGET_WIDTHS.FULL ]: [
-		'mdc-layout-grid__cell',
-		'mdc-layout-grid__cell--span-12',
-	],
-};
+import { WIDTH_GRID_COUNTER_MAP, WIDTH_GRID_CLASS_MAP } from './constants';
 
 /**
  * Adjusts class names to better fit into the current row knowing that the default sizes don't fill the row completely.
  *
  * @since 1.21.0
  *
- * @param {Array.<string>} classNames Current class names.
- * @param {number}         counter    Current counter.
- * @return {Array} Array where the first element is the updated list of class names
- *                 and the second element is the resulting counter after the update.
+ * @param {Array.<string>} classNames   Current class names.
+ * @param {Array.<number>} columnWidths Current column widths.
+ * @param {number}         counter      Current counter.
+ * @return {Array} Array where the first element is the updated list of class
+ *                 names, the second element is the updated list of column
+ *                 widths, and the third element is the resulting counter
+ *                 after the update.
  */
-function resizeClasses( classNames, counter ) {
+function resizeClasses( classNames, columnWidths, counter ) {
```

Comment:
Similar to a previous comment, it's a bit odd that this function handles grid classNames rather than grid cell sizes/props.

Maybe not worth changing right now but it seems like this would be a little less complicated if it were only dealing with widths rather than widths and class names.

Reply:
See above, that can be handled via #2671.